src/nspawn/nspawn.c

```diff
@@ -3311,6 +3348,12 @@ static int run(int master,
         if (r < 0)
                 return log_error_errno(errno, "Failed to install SIGCHLD handler: %m");
 
+        if (arg_network_namespace_path) {
+                netns_fd = open(arg_network_namespace_path, O_RDONLY|O_NOCTTY|O_CLOEXEC);
+                if (netns_fd < 0)
+                        return log_error_errno(errno, "cannot open file %s: %m", arg_network_namespace_path);
```

Comment:
hmm, could we do some minimal validation checking here, that the path actually refers to a netns? for example, call the NS_GET_NSTYPE ioctl on it and validate what is returns? (of course, this would need to be written in a defensive way, given that the ioctl was only added in 4.11: when we get ENOTTY or EOPNOTSUPP we should just accept the fd without further validation.

Reply:
maybe we can borrow some code from ip tool too ? in case NS_GET_NSTYPE ioct is not there..

Reply:
i wouldn't bother. I mean, this is just an extra safety net. It's fine if the additional checks only validate this on new kernels. Ultimately the setns() will fail anyway if we pass in wrong fds so strictly speaking the validation is not even necessary at all... Hence, let's do the safety check for current and future kernels, and let's just accept a tiny bit less validation for old kernels.